custom_components/sat/minimum_setpoint.py

```
@@ -1,87 +1,477 @@
+
+        if regime_state is None:
+            initial_minimum = self._initial_minimum_for_regime(new_regime_key=regime_key)
+            regime_state = RegimeState(minimum_setpoint=initial_minimum)
+            self._regimes[regime_key] = regime_state
+
+            _LOGGER.debug("Initialized regime %s with minimum_setpoint=%.1f from requested_setpoint=%.1f", regime_key, initial_minimum, requested_setpoint)
+
+        self._active_regime_key = regime_key
+        return regime_state
+
+    def _make_regime_key(self, cycles: "CycleStatistics", requested_setpoint: float, outside_temperature: float, areas_snapshot: Optional["AreasSnapshot"]) -> str:
```

Pylint (4.0.4)

[refactor] Too many branches (18/12)

(R0912)